### PR TITLE
Investment flow improvements

### PR DESCRIPTION
**Pull request by trival**
related https://github.com/Neufund/platform-frontend/issues/1274

### CODE REVIEW

File: intl/locales/en-en.json [hunk 1]
```diff
@@ -352,6 +352,8 @@
   "investment-flow.min-ticket-size": "MIN TICKET SIZE",
   "investment-flow.no-selected-wallet-warning": "Please select a wallet.",
   "investment-flow.select-wallet-and-currency": "select wallet and currency",
+  "investment-flow.success.congratulation-text": "Congratulations your investment transaction has been confirmed you can view your transaction on etherscan here. We will notify you when you can claim your Equity Tokens.",
```
Comment (desfero) [inline intl/locales/en-en.json]: `your transaction on etherscan here`, do we need to put a link to etherscan `here`?

File: app/modules/shared/tokenPrice/sagas.ts [hunk 2]
```diff
@@ -45,6 +44,8 @@ export async function loadTokenPriceDataAsync({
 }
 
 function* tokenPriceMonitor({ logger }: TGlobalDependencies): any {
+  const state: IAppState = yield select();
```
Comment (desfero) [inline app/modules/shared/tokenPrice/sagas.ts]: can we select by passing a selector function to select method to separate store shape from sagas?

File: app/components/modals/txSender/investment-flow/Investment.tsx [hunk 3]
```diff
@@ -139,7 +158,8 @@ function formatEth(val?: string | BigNumber): string | undefined {
 }
 
 export const InvestmentSelectionComponent = injectIntlHelpers((props: IProps & IIntlProps) => {
-  const { euroValue, ethValue, etherPriceEur, minTicketEur, intl, gasCostEth } = props;
+  const { euroValue, eto, ethValue, etherPriceEur, intl, gasCostEth } = props;
```
Comment (desfero) [inline app/components/modals/txSender/investment-flow/Investment.tsx]: can we destructure `props` inside function argument, as the current approach mixed two different access layers. For e.g. line 180 use `props.wallets` so basically the argument name `props` is used to access `walletInfo`, but on line 217 we use `eto` that is destructured from the same `props` object.

File: app/modules/shared/tokenPrice/sagas.ts (hunk 2 above)
Comment (Moejoe90) [reply app/modules/shared/tokenPrice/sagas.ts]: :+1:

File: intl/locales/en-en.json (hunk 1 above)
Comment (Moejoe90) [reply intl/locales/en-en.json]: Yes the link is easy to generate
As in 'http://etherscan.com/tx/transaction-hash

Comment (desfero) [reply intl/locales/en-en.json]: Do you know how correctly wrap `here` in a link tag with `react-intl`? Trying to find some docs for it but there are only docs for whole replacement of `{key}`.

File: app/modules/shared/tokenPrice/sagas.ts (hunk 2 above)
Comment (desfero) [reply app/modules/shared/tokenPrice/sagas.ts]: fixed

File: app/components/modals/txSender/investment-flow/Investment.tsx (hunk 3 above)
Comment (desfero) [reply app/components/modals/txSender/investment-flow/Investment.tsx]: fixed

File: intl/locales/en-en.json (hunk 1 above)
Comment (desfero) [reply intl/locales/en-en.json]: fixed

File: app/modules/shared/tokenPrice/sagas.ts [hunk 4]
```diff
@@ -44,8 +45,10 @@ export async function loadTokenPriceDataAsync({
 }
 
 function* tokenPriceMonitor({ logger }: TGlobalDependencies): any {
-  const state: IAppState = yield select();
-  if (!state.init.smartcontractsInit.done) return;
+  const isSmartContractInitDone: boolean = yield select(selectIsSmartContractInitDone);
```
Comment (Moejoe90) [inline app/modules/shared/tokenPrice/sagas.ts]: :+1:

File: app/components/shared/EtherscanLink.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import * as React from "react";
+
+import { CommonHtmlProps } from "../../types";
+
+interface IEtherscanTxLink {
+  txHash: string;
+}
+
+const EtherscanTxLink: React.SFC<IEtherscanTxLink & CommonHtmlProps> = ({
```
Comment (desfero) [inline app/components/shared/EtherscanLink.tsx]: @Moejoe90 @lizonp new component for external etherscan links. Two reasons why I have introduced it:
1. Don't want to have external link to different services hardcoded everywhere,
2. In future `EtherscanTxLink` and `EtherscanAddressLink` can be changed base on network we use (main/public(kaven,ropsten))

Comment (Moejoe90) [reply app/components/shared/EtherscanLink.tsx]: :+1:

**Pull request merged**